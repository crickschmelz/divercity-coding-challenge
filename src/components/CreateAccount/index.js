import React, { useState } from "react";
import { registerUser } from "../utils";
import {
  PageTitle,
  Button,
  Input,
  Form,
  Container,
  NoAccount,
  SignUpButton
} from "./style";

const CreateAccount = (props) => {
  const [username, setUserName] = useState("");
  const [password, setPassword] = useState("");
  const [name, setName] = useState("");

  const goToPage = (link) => {
    props.history.push(`/${link}`);
  };

  const clickLink = (e) => {
    e.preventDefault();
    const link = e.target.value;
    goToPage(link);
  };

  const register = () => {
    registerUser("https://divercity-test.herokuapp.com/register", {
      username,
      password,
      name
    }).then((data) => {
      return data;
    });
    goToPage("login");
  };

  return (
    <div>
      <Container>
        <PageTitle>Create Account</PageTitle>
        <Form onSubmit={register}>
          <Input
            type="text"
            required
            placeholder="Name"
            onChange={(event) => setName(event.target.value)}
          ></Input>
          <Input
            type="text"
            required
            placeholder="Username"
            onChange={(event) => setUserName(event.target.value)}
          ></Input>
          <Input
            type="text"
            required
            placeholder="Password"
            onChange={(event) => setPassword(event.target.value)}
          ></Input>
          <Button type="submit">SIGN UP</Button>
        </Form>
      </Container>
      <NoAccount>
        Already have an account?
        <SignUpButton onClick={clickLink} value="login">
          Sign Up
        </SignUpButton>
      </NoAccount>
    </div>
  );
};

export default CreateAccount;
