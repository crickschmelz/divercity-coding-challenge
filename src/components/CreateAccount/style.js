import styled from "styled-components";

export const NoAccount = styled.p`
  text-align: center;
  font-size: 12px;
`;

export const Container = styled.section`
  background-color: #ffffff;
  border-radius: 20px;
  margin: 10% 20% 0 20%;
  padding: 1% 0 5% 0;
`;

export const SignUpButton = styled.button`
  color: #3197e4;
  background-color: transparent;
  border: none;
  font-weight: 800;
  font-size: 12px;
  width: 60px;
  cursor: pointer;
`;

export const Form = styled.form`
  display: grid;
  justify-content: center;
  align-items: center;
`;

export const Input = styled.input`
  width: 200px;
  padding: 10px;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: 1px solid #bbb;
`;

export const PageTitle = styled.h3`
  text-align: center;
`;

export const Button = styled.button`
  color: #ffffff;
  background-color: #3197e4;
  border: none;
  font-weight: 800;
  font-size: 14px;
  width: 200px;
  padding: 10px;
  margin-top: 20px;
  border-radius: 30px;
  cursor: pointer;
`;
