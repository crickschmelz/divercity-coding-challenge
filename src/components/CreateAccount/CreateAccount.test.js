import React from "react";
import CreateAccount from "./CreateAccount";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<CreateAccount />).toJSON();
  expect(tree).toMatchSnapshot();
});
