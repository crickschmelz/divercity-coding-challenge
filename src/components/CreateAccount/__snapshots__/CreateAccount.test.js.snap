// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <section
    className="sc-AxirZ fBsFLx"
  >
    <h3
      className="sc-AxgMl ftqCsy"
    >
      Create Account
    </h3>
    <form
      className="sc-AxhCb cVhoPd"
      onSubmit={[Function]}
    >
      <input
        className="sc-AxhUy iwETNV"
        onChange={[Function]}
        placeholder="Name"
        required={true}
        type="text"
      />
      <input
        className="sc-AxhUy iwETNV"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
      />
      <input
        className="sc-AxhUy iwETNV"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="text"
      />
      <button
        className="sc-AxheI eCExDU"
        type="submit"
      >
        SIGN UP
      </button>
    </form>
  </section>
  <p
    className="sc-AxjAm jucUDx"
  >
    Already have an account?
    <button
      className="sc-AxiKw kgdTdl"
      onClick={[Function]}
      value="login"
    >
      Sign Up
    </button>
  </p>
</div>
`;
